src/Symfony/Component/Messenger/Transport/AmqpExt/AmqpTransport.php

```diff
@@ -58,7 +59,7 @@ public function send(Envelope $envelope): void
 
     private function getReceiver()
     {
-        return $this->receiver = new AmqpReceiver($this->connection, $this->serializer);
+        return $this->receiver = new InfiniteLoopReceiver(new AmqpReceiver($this->connection, $this->serializer));
```

Comment:
That's the only issue I can see here: it's more painful to create the receiver and it's very much error-prone for whoever is using these objects directly. Can't the `AmqpReceiver` use the `InfiniteLoopReceiver` instead? 🤔

Reply:
@sroze I'm not sure, the only way for Amqp to utilize this would be if we provided a utility method like:

```
function performOnInfiniteLoop(callable $shouldStop, callable $handler) {
  while (!$shouldStop()) {
     $handler();
  }
}
``` 

Also, are we 100% sure it's a bc break? because it's an internal class that's not really exposed to the outside since the TransportFactory is responsible for creating the actual receiver.